src/base/bittorrent/torrentinfo.cpp

```diff
@@ -305,10 +293,22 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
-void TorrentInfo::remapFiles(libtorrent::file_storage const &fileStorage)
+void TorrentInfo::stripRootFolder()
 {
-    if (!isValid()) return;
-    m_nativeInfo->remap_files(fileStorage);
+    if (filesCount() <= 1) return;
+
+    libtorrent::file_storage files = m_nativeInfo->files();
```

Comment:
Are you sure you didn't mean orig_files() here?
In any case, what is the problem that the lines below try to solve?